crates/forge/src/multi_runner.rs

```diff
@@ -391,8 +395,25 @@ impl MultiContractRunnerBuilder {
                 deployable_contracts.insert(id.clone(), (abi.clone(), bytecode, libs_to_deploy));
             }
 
-            if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
-                known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+            if !use_zk {
+                if let Some(bytes) = linked_contract.get_deployed_bytecode_bytes() {
+                    known_contracts.insert(id.clone(), (abi.clone(), bytes.to_vec()));
+                }
+            }
+        }
+        if use_zk {
+            let zk_contracts =
+                zk_output.unwrap().with_stripped_file_prefixes(root).into_artifacts();
+            for (id, contract) in zk_contracts {
+                let Some(abi) = contract.abi.as_ref() else {
+                    continue;
+                };
+                let Some(bytecode) = contract.bytecode.as_ref().and_then(|b| b.object.as_bytes())
```

Comment:
I think you can simplify this section a bit so you only have a single `.insert`, making the flow clearer